src/charm.py

```diff
@@ -236,8 +269,70 @@ def _on_synapse_pebble_ready(self, _: ops.HookEvent, charm_state: CharmState) ->
         Args:
             charm_state: The charm state.
         """
+        logger.debug("Found %d peer unit(s).", self.peer_units_total())
+        if charm_state.redis_config is None and self.peer_units_total() > 1:
+            logger.debug("More than 1 peer unit found. Redis is required.")
+            self.unit.status = ops.BlockedStatus("Redis integration is required.")
+            return
+        self.unit.status = ops.ActiveStatus()
```

Comment:
I am a little curious as to why the unit status is set to active before applying the changes.

Reply:
While scaling down, if there is only unit this will unblock it. The change_config then will call the https://github.com/canonical/synapse-operator/blob/main/src/charm.py#L123 set_unit_status if something goes wrong after that.

Reply:
gotcha, thanks for the info